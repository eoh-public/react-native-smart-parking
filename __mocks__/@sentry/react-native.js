export const setUser = jest.fn();
export const init = jest.fn();

export default {
  init,
  setUser,
};
