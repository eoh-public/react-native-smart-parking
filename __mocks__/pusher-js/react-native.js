const channel = {
  bind: jest.fn(),
};

const pusherClient = {
  subscribe: jest.fn(),
  unsubscribe: jest.fn(),
  disconnect: jest.fn(),
};

pusherClient.subscribe.mockImplementation(() => channel);

const MockPusher = jest.fn();
MockPusher.mockImplementation(() => pusherClient);

export default MockPusher;
