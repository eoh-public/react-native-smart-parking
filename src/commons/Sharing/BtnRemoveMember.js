import React, { memo, useCallback } from 'react';
import { TouchableOpacity, StyleSheet } from 'react-native';
import { IconOutline } from '@ant-design/icons-react-native';
import { Colors } from '../../configs';

const BtnRemoveMember = memo(({ member, onPressRemove }) => {
  const onPress = useCallback(() => {
    onPressRemove && onPressRemove(member);
  }, [member, onPressRemove]);
  return (
    <TouchableOpacity style={styles.buttonRemove} onPress={onPress}>
      <IconOutline name={'minus'} size={20} color={Colors.Gray8} />
    </TouchableOpacity>
  );
});

export default BtnRemoveMember;

const styles = StyleSheet.create({
  buttonRemove: {
    position: 'absolute',
    right: 0,
    top: 10,
    bottom: 10,
    paddingHorizontal: 10,
    justifyContent: 'center',
  },
});
