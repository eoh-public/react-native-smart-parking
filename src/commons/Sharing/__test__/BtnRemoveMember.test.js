import React from 'react';
import { TouchableOpacity } from 'react-native';
import { create, act } from 'react-test-renderer';
import BtnRemoveMember from '../BtnRemoveMember';

describe('Test BtnRemoveMember', () => {
  const member = {
    id: 1,
    name: 'Name',
    share_id: 2,
  };

  let wrapper;

  test('render BtnRemoveMember', () => {
    act(() => {
      wrapper = create(<BtnRemoveMember member={member} />);
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('onPressItem BtnRemoveMember', () => {
    const mockOnPressRemove = jest.fn();
    act(() => {
      wrapper = create(
        <BtnRemoveMember member={member} onPressRemove={mockOnPressRemove} />
      );
    });
    const instance = wrapper.root;
    const button = instance.findByType(TouchableOpacity);
    act(() => {
      button.props.onPress();
    });
    expect(mockOnPressRemove).toHaveBeenCalledTimes(1);
  });
});
