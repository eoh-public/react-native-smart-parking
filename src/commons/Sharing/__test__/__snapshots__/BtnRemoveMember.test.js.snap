// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BtnRemoveMember render BtnRemoveMember 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "bottom": 10,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 10,
      "position": "absolute",
      "right": 0,
      "top": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "antoutline",
          "fontSize": 20,
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    î¬€
  </Text>
</View>
`;
